src/main/java/org/moltimate/moltimatebackend/service/MotifTestService.java

```diff
@@ -0,0 +1,96 @@
+package org.moltimate.moltimatebackend.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.biojava.nbio.structure.Structure;
+import org.moltimate.moltimatebackend.dto.ActiveSiteAlignmentResponse;
+import org.moltimate.moltimatebackend.dto.MotifTestRequest;
+import org.moltimate.moltimatebackend.dto.PdbQueryResponse;
+import org.moltimate.moltimatebackend.model.Alignment;
+import org.moltimate.moltimatebackend.model.Motif;
+import org.moltimate.moltimatebackend.model.Residue;
+import org.moltimate.moltimatebackend.util.ProteinUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class MotifTestService {
+
+    @Autowired
+    private MotifService motifService;
+
+    @Autowired
+    private AlignmentService alignmentService;
+
+    public ActiveSiteAlignmentResponse testMotifAlignment(MotifTestRequest motifTestRequest) {
+        String motifPdbId = motifTestRequest.getPdbId();
+        String motifEcNumber = motifTestRequest.getEcNumber();
+        Structure motifStructure = motifTestRequest.motifStructure();
+        List<Residue> motifResidues = parseResidueEntries(motifTestRequest.getActiveSiteResidues());
+        Motif testMotif = motifService.generateMotif(motifPdbId, motifEcNumber, motifStructure, motifResidues);
+
+        List<Structure> structureList = new ArrayList<>();
+        List<String> failedIds = new ArrayList<>();
+        HashMap<String, List<Alignment>> results = new HashMap<>();
+
+        switch (motifTestRequest.getType()) {
+            case SELF:
+                structureList.add(motifStructure);
+                structureList.addAll(motifTestRequest.extractCustomStructuresFromFiles());
+
+                structureList.forEach(structure -> results.put(structure.getPDBCode(), new ArrayList<>()));
+                break;
+            case LIST:
+                PdbQueryResponse pdbQueryResponse = motifTestRequest.callPdbForResponse();
+                structureList.addAll(pdbQueryResponse.getStructures());
+                structureList.addAll(motifTestRequest.extractCustomStructuresFromFiles());
+
+                structureList.forEach(structure -> results.put(structure.getPDBCode(), new ArrayList<>()));
+                failedIds.addAll(pdbQueryResponse.getFailedPdbIds());
+                break;
+            case HOMOLOGUE:
+                // TODO: Using the given EC number find its HOMOLOGUE structures (from the PDB) and test alignment
+                break;
+            case RANDOM:
+                int max = motifTestRequest.getRandomCount();
+                while (max > 0) {
+                    String randomPdbId = RandomStringUtils.randomAlphanumeric(4).toUpperCase();
```

Comment:
Works for now, I guess. Might want to find a better way of doing this. Maybe we can get a list of all proteins somewhere and choose randomly from that

Reply:
I have been looking for one and when you consider that the PDB is always collecting more its a tough situation (since the RCSB doesn't have API support for this), I'd be ok with collecting 100 or so that work and then including those in a DB entry and then we can just pull randomly from there.

Reply:
This sounds like it might be a good idea. Would be cool to get multiple from a variety of EC numbers

Reply:
I will make an issue to deal with using `https://www.rcsb.org/pdb/rest/getCurrent` which gives the full list. The request takes ~13-15 seconds right now though so we shouldn't run it regularly.